src/Parallel/Callback.hpp

```diff
@@ -97,6 +97,71 @@ class SimpleActionCallback<SimpleAction, Proxy> : public Callback {
   std::decay_t<Proxy> proxy_{};
 };
 
+/// Wraps a call to a threaded action and its arguments.
+/// Can be invoked only once.
+template <typename ThreadedAction, typename Proxy, typename... Args>
+class ThreadedActionCallback : public Callback {
```

Comment:
We have a test for these in `Unit/Parallel/Test_Callback.cpp`. Could you add some there? I know it's a bit more work, but just so we are guaranteed that they are working

Reply:
Not a problem! Done!